app/component/itinerary/NaviCardContainer.js

```diff
@@ -98,6 +85,7 @@ function NaviTop(
       if (legChanged) {
         focusToLeg?.(newLeg);
         setCurrentLeg(newLeg);
+        setCardExpanded(false);
       }
 
       if (incomingMessages.size || legChanged) {
```

Comment:
This code block is in wrong level.  Currentleg and newLeg ban be null while waiting, stiull new messages may appear.

Also  if (!focusRef.current && focusToLeg) {}  block is under wrong if block

Reply:
Valid point! Will fix this!